Test/Dynamics/test_dynamics.py

```diff
@@ -52,17 +55,22 @@ def rhs_real(t, x):
     assert overlap(driver.state, psi_scipy) == approx(1.0)
 
 
-def test_imag_time_evolution():
-    hi, ham, ts, psi0 = _setup_model()
-    ham_matrix = ham.to_sparse()
+def test_python_imag_time_evolution():
+    ham, psi0 = _setup_model()
+    ham_matrix = ham.to_linear_operator()
 
-    driver = ExactTimePropagation(
-        ham, ts, t0=0.0, initial_state=psi0, propagation_type="imaginary"
+    driver = PyExactTimePropagation(
```

Comment:
@femtobit just a reminder here that we should change this to ExactTimePropagation without Py, I can do that later

Reply:
I remember us discussing leaving it wit `Py` and having `ExactTimePropagation` throw an error if used, as the classes are not compatible without (small) changes.

Reply:
Although this PR does contain quite a few other backwards-incompatible changes anyway

Reply:
Yes, unfortunately this is almost inevitable given the amount of changes... In reality this should be the actual v2.0 release, not v2.1... but I think it should be fine

Reply:
It could arguably be 3.0, even.

Reply:
Yes, it might make sense actually... but then in that case I'd remove all C++ already. Actually that makes sense :)

Reply:
Well, there isn't too much C++ left now in any case.

Reply:
Yes indeed, only a few machines left to be rewritten + the optimizers and the graphs... I can easily rewrite all the optimizers in python, but I am a bit worried by rewriting Lattice in python... let's see